### PR TITLE
Assorted infrastructure enhancements

### PR DESCRIPTION
**Pull request by rsimha**
**PR highlights:**
- Get rid of the root dependency on `fancy-log` (it's part of `gulp` and isn't being maintained)
- Add `.vscode/settings.json` to enable auto-fix-on-save via the VSCode linter (if installed)
- Add a utilities file `build-system/utils.js` and move `timedExecOrDie()` to it
- Add a minimal `printChangeSummary()` function to print a change log (distilled version of what's in `amphtml`)
- Consolidate package installation and setup in `install_dependencies.sh` (necessary for change summary printing)
- Update process spawning functions in `exec.js` (current version of `spawnSync` was buggy on GH actions)
- Copy minimal versions of the necessary utils from the `amphtml` repo to `ci.js` and `git.js`

**Screenshot:**

image

Addresses https://github.com/ampproject/amp-github-apps/pull/1284#issuecomment-831404707

### CODE REVIEW

File: build-system/pr-check.js [hunk 1]
```diff
@@ -16,33 +16,10 @@
 'use strict';
 
 const {APPS_TO_TEST, determineBuildTargets} = require('./build-targets');
-const {cyan, green} = require('kleur/colors');
-const {execOrDie} = require('./exec');
+const {cyan} = require('kleur/colors');
```
Comment (rileyajones) [inline build-system/pr-check.js:19]:
```suggestion
const {cyan} = require('./common/colors');
```

File: build-system/utils.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  gitBranchCreationPoint,
+  gitBranchName,
+  gitCommitHash,
+  gitDiffCommitLog,
+  gitDiffStatMain,
+  gitMergeBaseMain,
+  shortSha,
+} = require('./git');
+const {ciPullRequestSha, isCiBuild} = require('./ci');
+const {cyan, green} = require('kleur/colors');
```
Comment (rileyajones) [inline build-system/utils.js:28]:
```suggestion
const {cyan} = require('./common/colors');
```

File: build-system/pr-check.js (hunk 1 above)
Comment (rsimha) [reply build-system/pr-check.js:19]: This is the `amp-github-apps` repo. There's no `common/` directory 😃

File: build-system/utils.js (hunk 2 above)
Comment (rsimha) [reply build-system/utils.js:28]: Same comment as above.

**Review by rileyajones — APPROVED** [review]
Couple minor comments, overall LGTM

**Comment by rsimha** [comment]
Confirmation that this works as desired for new PRs: see [logs](https://github.com/ampproject/amp-github-apps/runs/2579872408?check_suite_focus=true)

image

**Pull request merged**